### PR TITLE
Add amp-wistia-player in order to load Wistia videos

### LINKED ISSUE #12644: Intent to Implement: amp-wistia-player
Proposed `amp-wistia-player` component would embed a responsive Wistia player in an iframe. This single component would support publishers using Wistia.

Attributes specifying the `media hashed-id` will determine the embed URI that will be used. 

Example:

```
<amp-wistia-player 
    data-media-hashed-id="u8p9wq6mq8"
    width="512" 
    height="360">
</amp-wistia-player>
```

Above would yield http://fast.wistia.net/embed/iframe/u8p9wq6mq8

The component supports using play, pause, mute, unmute postMessage event to interact with the video.

### PR DESCRIPTION
**Pull request by chen-anders**
Closes #12644 .

### CODE REVIEW

File: extensions/amp-wistia-player/validator-amp-wistia-player.protoascii [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#
+# Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (honeybadgerdontcare) [inline extensions/amp-wistia-player/validator-amp-wistia-player.protoascii]: replace 2017 with 2018

**Comment by honeybadgerdontcare** [comment]
validation changes look good.

**Comment by chen-anders** [comment]
Signed the CLA.

File: extensions/amp-wistia-player/0.1/amp-wistia-player.js [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+      if (state === 'playing') {
+        this.element.dispatchCustomEvent(VideoEvents.PLAYING);
+      } else if (state === 'paused') {
+        this.element.dispatchCustomEvent(VideoEvents.PAUSE);
+      } else if (state === 'ended') {
+        this.element.dispatchCustomEvent(VideoEvents.PAUSE);
+        this.element.dispatchCustomEvent(VideoEvents.ENDED);
+      }
+    } else if (data['method'] == 'mutechange') {
+      isMuted = (data['args'] ? data['args']['_isMuted'] : undefined)
```
Comment (cathyxz) [inline extensions/amp-wistia-player/0.1/amp-wistia-player.js]: `const isMuted = `?

File: examples/wistiaplayer.amp.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  <head>
+    <meta charset="utf-8">
+    <link rel="canonical" href="amps.html">
+    <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+    <style>body {opacity: 0}</style><noscript><style>body {opacity: 1}</style></noscript>
+    <script async custom-element="amp-wistia-player" data-amp-report-test="amp-wistia-player" src="https://cdn.ampproject.org/v0/amp-wistia-player-0.1.js"></script>
+    <script async src="https://cdn.ampproject.org/v0.js"></script>
+  </head>
+  <body>
+    <amp-wistia-player
```
Comment (aghassemi) [inline examples/wistiaplayer.amp.html]: needs a `layout`

File: extensions/amp-wistia-player/0.1/amp-wistia-player.js [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+  /** @override */
+  layoutCallback() {
+    const mediaId = user().assert(
+        this.element.getAttribute('data-media-hashed-id'),
+        'The data-media-hashed-id attribute is required ' +
+            'for <amp-wistia-player> %s',
+        this.element);
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    iframe.setAttribute('title', 'Wistia Video Player');
```
Comment (aghassemi) [inline extensions/amp-wistia-player/0.1/amp-wistia-player.js]: page author should control this. Maybe 
```
iframe.setAttribute('title', this.element.getAttribute('title') || 'Wistia Video Player');
```

File: extensions/amp-wistia-player/0.1/amp-wistia-player.js [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('scrolling', 'no');
+    iframe.setAttribute('allowtransparency', '');
+    iframe.setAttribute('allowfullscreen', 'true');
+    iframe.src = '//fast.wistia.net/embed/iframe/' + encodeURIComponent(
```
Comment (aghassemi) [inline extensions/amp-wistia-player/0.1/amp-wistia-player.js]: no relative protocol, must be `https://`

**Comment by aghassemi** [comment]
@chen-anders To fix the Travis CI failure, you players needs to be added to the whitelist file here: https://github.com/ampproject/amphtml/blob/master/build-system/dep-check-config.js#L175

File: examples/wistiaplayer.amp.html (hunk 3 above)
Comment (chen-anders) [reply examples/wistiaplayer.amp.html]: added here: https://github.com/ampproject/amphtml/pull/12654/commits/5838fe1fb6b9439cac1cf90725013fd09bbc8303

**Comment by chen-anders** [comment]
CI failures have been resolved.

**Comment by chen-anders** [comment]
Is there some additional work that needs to be done in order to be listed on the following page: https://www.ampproject.org/docs/reference/components#media ?

**Comment by aghassemi** [comment]
@chen-anders We periodically update that page. You can also submit a PR against this file: https://github.com/ampproject/docs/blob/e7696149ce0c231525e07a8ea2a099e1a3a23a99/content/docs/reference/components.md to get it there faster.

**Pull request merged**